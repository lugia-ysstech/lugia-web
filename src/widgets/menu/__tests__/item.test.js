//@flow

import * as React from 'react';
import chai from 'chai';
import MenuItem from '../item';
import 'jest-styled-components';

import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

const { expect: exp } = chai;
const { mockFunction, VerifyOrder, VerifyOrderConfig } = require('@lugia/jverify');

describe('Item', () => {
  it('children: null', () => {
    const target = renderer.create(<MenuItem />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('Single MenuItem', () => {
    const defaultItem = renderer.create(<MenuItem>hello</MenuItem>).toJSON();
    const target = renderer.create(<MenuItem>hello</MenuItem>).toJSON();
    expect(target).toMatchSnapshot();
    expect(target).toMatchObject(defaultItem);
  });

  it('Single MenuItem checked', () => {
    const target = renderer.create(
      <MenuItem checked>
        <a>hello</a>
      </MenuItem>
    );
    expect(target).toMatchSnapshot();
  });
  it('Mutliple MenuItem', () => {
    const defaultItem = renderer
      .create(
        <MenuItem mutliple>
          <a>hello</a>
        </MenuItem>
      )
      .toJSON();
    const target = renderer
      .create(
        <MenuItem mutliple checked={false}>
          <a>hello</a>
        </MenuItem>
      )
      .toJSON();
    expect(target).toMatchSnapshot();
    expect(target).toMatchObject(defaultItem);
  });

  it('Mutliple MenuItem checked', () => {
    const target = renderer.create(
      <MenuItem mutliple checked>
        <a>hello</a>
      </MenuItem>
    );
    expect(target).toMatchSnapshot();
  });
  it('onClick', () => {
    const order = VerifyOrder.create();
    const mockClick = mockFunction.create(VerifyOrderConfig.create('eventHandle', order));
    const onClick = mockClick.getFunction();
    const target = mount(<MenuItem onClick={onClick} />);
    target.find('li').simulate('click', {});
    order.verify(obj => {
      const { eventHandle } = obj;
      eventHandle(VerifyOrder.Object);
    });
  });
});
