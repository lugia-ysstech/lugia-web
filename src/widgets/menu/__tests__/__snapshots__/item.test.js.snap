// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Mutliple MenuItem 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  font-weight: 100;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition-property: background-color,border,borderRadius,opacity,boxShadow;
  transition-property: background-color,border,borderRadius,opacity,boxShadow;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  height: 3.6rem;
}

.c0 > i {
  vertical-align: middle;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  overflow: hidden;
  padding-top: 0rem;
  padding-bottom: 0rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c1 i {
  vertical-align: middle;
}

.c2 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 500;
}

<li
  checked={false}
  className="c0"
  disabled={false}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "color": "#363B40",
    }
  }
  title=""
>
  <div
    className="c1"
    style={
      Object {
        "height": "3.2rem",
      }
    }
  >
    <span
      className="c2"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      <a>
        hello
      </a>
    </span>
  </div>
</li>
`;

exports[`Item Mutliple MenuItem checked 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  font-weight: 100;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition-property: background-color,border,borderRadius,opacity,boxShadow;
  transition-property: background-color,border,borderRadius,opacity,boxShadow;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  height: 3.6rem;
}

.c0 > i {
  vertical-align: middle;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  overflow: hidden;
  padding-top: 0rem;
  padding-bottom: 0rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c1 i {
  vertical-align: middle;
}

.c2 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 500;
}

<li
  checked={true}
  className="c0"
  disabled={false}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "color": "#363B40",
    }
  }
  title=""
>
  <div
    className="c1"
    style={
      Object {
        "height": "3.2rem",
      }
    }
  >
    <span
      className="c2"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      <a>
        hello
      </a>
    </span>
  </div>
</li>
`;

exports[`Item Single MenuItem 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  font-weight: 100;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition-property: background-color,border,borderRadius,opacity,boxShadow;
  transition-property: background-color,border,borderRadius,opacity,boxShadow;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  height: 3.6rem;
}

.c0 > i {
  vertical-align: middle;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  overflow: hidden;
  padding-top: 0rem;
  padding-bottom: 0rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c1 i {
  vertical-align: middle;
}

.c2 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 500;
}

<li
  checked={false}
  className="c0"
  disabled={false}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "color": "#363B40",
    }
  }
  title="hello"
>
  <div
    className="c1"
    style={
      Object {
        "height": "3.2rem",
      }
    }
  >
    <span
      className="c2"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      hello
    </span>
  </div>
</li>
`;

exports[`Item Single MenuItem checked 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  font-weight: 100;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition-property: background-color,border,borderRadius,opacity,boxShadow;
  transition-property: background-color,border,borderRadius,opacity,boxShadow;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  height: 3.6rem;
}

.c0 > i {
  vertical-align: middle;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  overflow: hidden;
  padding-top: 0rem;
  padding-bottom: 0rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c1 i {
  vertical-align: middle;
}

.c2 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 500;
}

<li
  checked={true}
  className="c0"
  disabled={false}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "color": "#363B40",
    }
  }
  title=""
>
  <div
    className="c1"
    style={
      Object {
        "height": "3.2rem",
      }
    }
  >
    <span
      className="c2"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      <a>
        hello
      </a>
    </span>
  </div>
</li>
`;

exports[`Item children: null 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  font-weight: 100;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition-property: background-color,border,borderRadius,opacity,boxShadow;
  transition-property: background-color,border,borderRadius,opacity,boxShadow;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  height: 3.6rem;
}

.c0 > i {
  vertical-align: middle;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  overflow: hidden;
  padding-top: 0rem;
  padding-bottom: 0rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c1 i {
  vertical-align: middle;
}

.c2 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 500;
}

<li
  checked={false}
  className="c0"
  disabled={false}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "color": "#363B40",
    }
  }
  title=""
>
  <div
    className="c1"
    style={
      Object {
        "height": "3.2rem",
      }
    }
  >
    <span
      className="c2"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    />
  </div>
</li>
`;
